app/models/template.rb

```diff
@@ -31,8 +31,9 @@ def self.template_includes
     []
   end
 
-  def preview_host_collection
-    Host.authorized(:view_hosts).order(:name).limit(100)
```

Comment:
I think, this should be handled by the plugin itself. If the plugin needs a limit, it can easily use ```preview_host_collection.limit(100)```.